### DISCUSSION TITLE
使用trieve时编译错误

### DISCUSSION THREAD

**Discussion by allentown521** [fuma-nama/fumadocs#1179] [Q&A]
都是最新版本：
    "fumadocs-core": "14.6.2",
    "fumadocs-mdx": "11.1.2",
    "fumadocs-ui": "14.6.2",
    "memoize-one": "^5.1.0",
    "next": "15.0.4",
    "react": "19.0.0",
    "react-dom": "19.0.0",
    "supports-color": "^10.0.0",
    "trieve-fumadocs-adapter": "^1.2.4",

 ⨯ ./node_modules/trieve-fumadocs-adapter/dist/components/dialog/search-trieve.mjs
Attempted import error: 'TagsList' is not exported from 'fumadocs-ui/components/dialog/search' (imported as 'TagsList').

查看了 packages/ui/src/components/dialog/search.tsx 源代码确实没有导出 TagsList，所以是需要更新trieve-fumadocs-adapter的代码么？

**Comment by fuma-nama** [comment]
Cannot help with that, it is a community adapter. You can use the default one powered by Orama, or Algolia with official support

**Comment by allentown521** [comment]
好的，我刚才尝试切换到了
algolia（因为默认的orama不支持中文搜索），英文搜索运行良好。但是我的docs下面是有简体中文和繁体中文的mdx，我发现并没有被同步到algolia的索引中，我需要做什么调整么？这是代码
[image: image.png]

Fuma Nama ***@***.***> 于2024年12月20日周五 16:06写道：

**Comment by allentown521** [comment]
解决了：
[image: image.png]

Focus Apps ***@***.***> 于2024年12月20日周五 16:13写道：

**Reply by fuma-nama** [reply]
Nice to hear that, btw for Chinese, we do have support on Orama.

See https://fumadocs.vercel.app/docs/headless/search/orama#special-languages, note that we use Orama v2 as v3 has some issues with external tokenizers

**Comment by allentown521** [comment]
我有过尝试，但是遇到了问题：[image: image.png]
[image: image.png]

Fuma Nama ***@***.***> 于2024年12月20日周五 16:45写道：

**Comment by allentown521** [comment]
补充一点是，我是在你们的i18n demo上修改的，包含两个文件，文档只涉及了route.ts，不知道和这有关系么？[image: image.png]
[image: image.png]

Focus Apps ***@***.***> 于2024年12月20日周五 17:11写道：

**Reply by fuma-nama** [reply]
Hmm I cannot see the images

**Reply by allentown521** [reply]
![image](https://github.com/user-attachments/assets/a57637dc-9332-4883-a61e-c49cee7aea87)
![image](https://github.com/user-attachments/assets/f5fdd9e7-3bf1-4857-8b96-219ad30d2e86)
![image](https://github.com/user-attachments/assets/f43d07e0-30a4-49d8-9cbc-0a23748b1d4b)
![image](https://github.com/user-attachments/assets/b3472675-4566-421e-aafc-2d6be9b685ad)
之前都是gmail里面直接回复的，可能没上传完就发出去了

**Comment by fuma-nama** [comment]
The `cn` is the locale code you're using, it should be `zh-CN` in your case. And assuming it's Orama v2, it takes some time to compile/load tokenizers, but usually works. (Otherwise, you'll need 3rd party solutions like Algolia, running large tokenizers on local can be slow)

`route-full.ts` is only for code examples, it won't affect anything.

**Comment by allentown521** [comment]
我不知道是不是响应慢的原因，我等待了可能有几分钟了，期间网页无法响应任何事件了。但是我只要加上了zh-CN,即使我搜索英文也是一样无响应。不过如果你确定是这个原因，那可能只能换在线方案了。文档估计30多页的样子

Fuma Nama ***@***.***> 于 2024年12月20日周五 18:48写道：

**Comment by allentown521** [comment]
我没有修改 orama 版本，所以它应该是v2版本

Focus Apps ***@***.***> 于 2024年12月20日周五 19:24写道：

**Reply by fuma-nama** [reply]
Check `package.json` and make sure Orama dependencies are all v2

If that doesn't work, I think Algolia will be the way to go. I'm thinking to improve the guide about i18n search during Xmas holiday

**Comment by allentown521** [comment]
确认是v2版本：
![image](https://github.com/user-attachments/assets/c01135d7-b356-4428-8db5-5a7aebd22a77)

**Comment by allentown521** [comment]
![image](https://github.com/user-attachments/assets/83f2fae9-c1a2-4319-9cfc-bb5da9a39c60)

**Reply by fuma-nama** [reply]
Hmm I see, you probably have to use Algolia for now, it supports multilingual out-of-the-box.

Regarding the Orama i18n issue, this is from my Macbook:

![image](https://github.com/user-attachments/assets/97859569-a923-4a2b-91b1-41d1b1432e25)

It took ~15 seconds for initial query.
I think it's indeed due to the high resource requirement to tokenize Chinese content, cloud solutions would be better
